internal/httprouter/config.go

```diff
@@ -0,0 +1,70 @@
+package router
+
+import (
+	"boilerplate/internal/framework"
+	"context"
+	"fmt"
+	"github.com/spf13/viper"
+	"go.uber.org/dig"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+type ModuleConfig struct {
+	Port                   int `mapstructure:"HTTP_APP_PORT"`
+	RedirectTrailingSlash  bool
+	RedirectFixedPath      bool
+	HandleMethodNotAllowed bool
+	HandleOPTIONS          bool
+	NotFound               http.Handler
+	MethodNotAllowed       http.Handler
+	PanicHandler           func(http.ResponseWriter, *http.Request, interface{})
+	container              *dig.Container
+}
+
+func ModuleHooks(
+	lc fx.Lifecycle,
+	router *Router,
+	routes *framework.Routes,
+	logger *zap.Logger,
+) error {
+	lc.Append(
+		fx.Hook{
+			OnStart: func(context.Context) error {
+				router.AddRoutes(routes.GetRoutesInfo())
+				logger.Info(fmt.Sprintf("Listen to the port: %d", router.port))
+				go http.ListenAndServe(fmt.Sprintf(":%d", router.port), router.router)
+				return nil
+			},
+			OnStop: func(ctx context.Context) error {
+				logger.Info("Stopping http-server")
+				return nil
+			},
+		},
+	)
+
+	return nil
+}
+
+func HttpRouterModule(config ModuleConfig) fx.Option {
```

Comment:
Then it is possible to start module with default settings

Reply:
```suggestion
func NewModule() fx.Option {
  defaultConfig := &ModuleConfig{
       key: "defaultValue"
  }
  
  return NewModuleWithConfig(defaultConfig)
}

func NewModuleWithConfig(config *ModuleConfig) fx.Option {
```

Reply:
Changing the name HttpRouterModule to NewModule - it's ok. But hiding the config inside the module - is not good for me. How can I change, for example, the PanicHandler callback? It cannot be defined in ENV or yaml.